scripts/linter_utils.py

```diff
@@ -68,12 +72,23 @@ def get_from_cache(key, factory):
             *. The result of factory(), or the last value to be associated to
             key.
         """
-        if key in cache:
+        if key not in cache:
+            with threadsafe_access(key):
+                if key not in cache:
+                    # We need to use a container because variable assignment
```

Comment:
I don't really understand what's going on here. Can you add a link to a reference (and perhaps more elaboration on the explanation)?